### PR TITLE
fix: send origin in query tagging

### PR DESCRIPTION
**Pull request by lauramargar**
# Pull request template
 
When you click on a brand recommendation, the origin param is not included in the query tagging. That's because the `BaseEventButton` used on the `QueryPreviewButton` component emits `UserAcceptedAQueryPreview` event without any feature param in the metadata so that the origin couldn't be created.

To fix it, we are going to pass the origin through a prop and it will be added to the event metadata.

## Motivation and context

- [ ] Dependencies. If any, specify:
- [X] Open issue. If applicable, link: [RST-1853](https://searchbroker.atlassian.net/browse/RST-1853)

## Type of change

- [X] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that causes existing functionality to not work as expected)
- [ ] Change requires a documentation update

## What is the destination branch of this PR?

- [X] `Main`
- [ ] Other. Specify:

## How has this been tested?

Click on a brand recommendation and check if the origin `customer:no_query` is sent in the query tagging.

## Checklist:

- [ ] My code follows the **[style guidelines](./CONTRIBUTING.md#style-guides)** of this project.
- [ ] I have performed a **self-review** on my own code.
- [ ] I have **commented** my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the **[documentation](./CONTRIBUTING.md#documentation-style-guide)**.
- [ ] My changes generate **no new warnings**.
- [ ] I have added **tests** that prove my fix is effective or that my feature works.
- [ ] New and existing **unit tests pass locally** with my changes.

[RST-1853]: https://searchbroker.atlassian.net/browse/RST-1853?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by albertjcuac — DISMISSED** [review]

**Comment by herrardo** [comment]
Germán 👀

File: packages/x-components/src/components/base-event-button.vue [hunk 1]
```diff
@@ -26,6 +27,14 @@
       events: {
         type: Object as PropType<Partial<XEventsTypes>>,
         required: true
+      },
+      /**
+       * The origin property for the request on each query preview.
+       *
+       * @public
+       */
+      queryFeature: {
```
Comment (diegopf) [inline packages/x-components/src/components/base-event-button.vue:36]: I think would be better to not limit this to the queryFeature. I'd something like:

```ts
      metadata: {
        type: Object as PropType<Omit<WireMetadata, 'moduleName'>>
      }
...
      function emitEvents() {
        Object.entries(props.events).forEach(([event, payload]) =>
          $x.emit(event as XEvent, payload, { target: rootRef.value, ...props.metadata })
        );
      }
```
I'm not sure if we could have issues with the wiring types. If we do, probably we could even relax them here.

File: packages/x-components/src/components/base-event-button.vue [hunk 2]
```diff
@@ -38,7 +47,7 @@
        */
       function emitEvents() {
         Object.entries(props.events).forEach(([event, payload]) =>
-          $x.emit(event as XEvent, payload, { target: rootRef.value })
+          $x.emit(event as XEvent, payload, { target: rootRef.value, feature: props.queryFeature })
```
Comment (diegopf) [inline packages/x-components/src/components/base-event-button.vue:50]: See comment above

**Review by diegopf — CHANGES_REQUESTED** [review]

File: packages/x-components/src/components/base-event-button.vue (hunk 1 above)
Comment (lauramargar) [reply packages/x-components/src/components/base-event-button.vue:36]: You're right!! 👍🏽

**Review by diegopf — APPROVED** [review]

**Review by albertjcuac — APPROVED** [review]

**Pull request merged**